### DISCUSSION TITLE
Questions about prefixes in OT network

### DISCUSSION THREAD

**Discussion by kderda** [openthread/openthread#10032] [Q&A] [answered]
Hello,
I am playing with OT prefixes and I came up with a few questions and also would like to ensure that my understanding is correct:
1. OMR prefixes - 64-bit prefixes which can be ULA or GUA prefixes, can be set manually, generated by Border Router, or delegated with DHCPv6. Regardless of the source this prefix must be different from Backbone Link's prefix (regarding to "Border Router Best Practices" whitepaper) and propagated as RIO option with Router Advertisements. Is that correct?
2. Domain Prefixes - shared by few Thread networks but also different from the Backbone Link's prefix. Is that correct?
3. Border Routers should not respond to Neighbor Solicitation messages except of DUA addresses (as ND Proxy). Is that correct?
4. Obtaining OMR prefix with DHCPv6 Prefix Delegation - Border Router's backbone interface has to request a prefix on behalf of Thread interface and pass it as a Router Advertisement transmitted that would be transmitted to Thread network (based on current implementation). This could be done by simply setting such prefix with C API after DHCPv6 reponse reception, right?
5. What is an example use case of prefix with `On-Mesh` flag set to `false`?

**Comment by jwhui** [comment, marked as answer]
> OMR prefixes - 64-bit prefixes which can be ULA or GUA prefixes, can be set manually, generated by Border Router, or delegated with DHCPv6. Regardless of the source this prefix must be different from Backbone Link's prefix (regarding to "Border Router Best Practices" whitepaper) and propagated as RIO option with Router Advertisements. Is that correct?

Yes, correct.

> Domain Prefixes - shared by few Thread networks but also different from the Backbone Link's prefix. Is that correct?

Yes, correct.

> Border Routers should not respond to Neighbor Solicitation messages except of DUA addresses (as ND Proxy). Is that correct?

Border Routers only perform ND Proxy for the Domain Prefix.

> Obtaining OMR prefix with DHCPv6 Prefix Delegation - Border Router's backbone interface has to request a prefix on behalf of Thread interface and pass it as a Router Advertisement transmitted that would be transmitted to Thread network (based on current implementation). This could be done by simply setting such prefix with C API after DHCPv6 reponse reception, right?

The current OpenThread implementation uses a RA message to receive the delegated prefix.

There is also https://github.com/openthread/openthread/pull/10000, which also introduces an API to provide the delegated prefix.

> What is an example use case of prefix with On-Mesh flag set to false?

The on-mesh flag is intended to mirror the RFC 4861 on-link flag. It is intended to enable use cases where all packets destined to the prefix should be forwarded to the router.

**Reply by kderda** [reply]
@jwhui Thank you for the answers
